packages/core/aqua-compiler/src/generate/index.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CompilationResult, JSTypeGenerator, OutputType, TSTypeGenerator } from './interfaces.js';
+import { getPackageJsonContent } from '../utils.js';
+import { ServiceGenerator } from './service.js';
+import { FunctionGenerator } from './function.js';
+import header from './header.js';
+
+export async function generateSources({ services, functions }: CompilationResult, outputType: OutputType) {
+    const typeGenerator = outputType === 'js' ? new JSTypeGenerator() : new TSTypeGenerator();
+    const { version, dependencies } = await getPackageJsonContent();
+    return `/* eslint-disable */
+// @ts-nocheck
+${header(version, dependencies['@fluencelabs/aqua-api'], outputType)};
+
+// Services
+${new ServiceGenerator(typeGenerator).generate(services)}
+
+// Functions
+${new FunctionGenerator(typeGenerator).generate(functions)}
+
+/* eslint-enable */
```

Comment:
if here is now a need for having header in two places - I would still make this function local to this module
header function is not intended to be used anywhere else, so it can be placed right here in this file

also `/* eslint-disable */` and `// @ts-nocheck` can be moved there

also `/* eslint-enable */`, I think, is redundant, it was previously there, I know, but I think it's not needed since we disable the stuff for the whole file

Reply:
Not agree with your separation rules. I have my own separation rule.

Reply:
Comment regarding eslint declarations sounds reasonable.

Reply:
what rule do you have? I think having less files in most cases is good. Less file navigation. Especially if it's just a small helper function that is intended to be used in one place exclusivly - IMHO doesn't deserve a separate file. But you can leave it as it is if you have a strong preference/rule against my points, I just wanna hear them for future reference, if you don't mind. Would be great to agree on this

Reply:
Well, in my setup, i split entities(services, function, headers, etc...) in each separate file. For now header is a small utility function, later it could grow. I like consistency - if i use separation for at least 1 entity, then i would prefer to use it for every other one, even the one containing 1 line of code.

Reply:
what you consider to be an entity?
I don't see much consistency in your code in this regard in particular since there several places where you happily have multiple functions in one file and it doesn't cause any problems. e.g. one could argue that `genTypeName` and `typeToTs` are separate enteties and should be indeffirent files (still don't know why one would argue for that and what is an entity)
I doubt you will find many people who would consider a small utility function used in one place as an entity

Reply:
in the current context, entity is a building block in file generation(services, function, headers, etc...)

Reply:
I would appreciate if this is at least not an anonymouse function, if we agree that exported functions are eastier to use when they are named